resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -397,6 +407,64 @@ private[spark] class KubernetesClusterSchedulerBackend(
     }
   }
 
+  /*
+ * KubernetesAllocator class watches all the executor pods associated with
+ * this SparkJob and creates new executors when it is appropriate.
+ */
+  private[spark] class KubernetesAllocator(
+    client: KubernetesClient,
+    interval: Int,
+    backend: KubernetesClusterSchedulerBackend) extends Logging {
+
+    private val scheduler = Executors.newScheduledThreadPool(1)
+    private val allocatorRunnable: Runnable = new Runnable {
```

Comment:
instead of using Java `Runnable` and specifically submitting the task, consider using Scala's `ExecutionContext` and `Future` blocks.